server/issue.go

```diff
@@ -715,6 +715,42 @@ func (p *Plugin) getIssueAsSlackAttachment(ji Instance, jiraUser JIRAUser, issue
 	return parseIssue(issue), nil
 }
 
+func (p *Plugin) unassignJiraIssue(mmUserId, issueKey string) (string, error) {
+	ji, err := p.currentInstanceStore.LoadCurrentJIRAInstance()
+	if err != nil {
+		return "", err
+	}
+
+	jiraUser, err := ji.GetPlugin().userStore.LoadJIRAUser(ji, mmUserId)
+	if err != nil {
+		return "", err
+	}
+
+	client, err := ji.GetClient(jiraUser)
+	if err != nil {
+		return "", err
+	}
+
+	// check for valid issue key
+	_, err = client.GetIssue(issueKey, nil)
```

Comment:
Why the GetIssue? The transition will fail if the issue doesn't exist or is privileged. Ditto for `assign`?

Reply:
The spec that was asked in the ticket was to provide two different posts.  

1) "You do not have the appropriate permissions to perform this action. Please contact your Jira administrator.
2) "We couldn't find the issue key. Please confirm the issue key and try again."

The response from a failed transition does fail with "Issue does not exist or you do not have permission to see it.", but that didn't satisfy the requested ticket request.

Reply:
So in this case I don't think it's critical, but if we are doing it only to improve the error reporting, we should be doing it only in the case of an error (i.e. try to transition first, and if it fails - diagnose to get the exact reason for it. I'm ok with it as is since load/performance are not a huge concern for command.

Reply:
Yes, that makes sense and is similar to a pythonic principle "ask for forgiveness, not permission".  In this case to save an additional `getIssue()` call, "transition first (without permission) and if error occurs, (ask forgiveness) and handle the error.